packages/service-library/src/servicelib/file_utils.py

```diff
@@ -4,6 +4,9 @@
 
 async def remove_directory(path: Path, only_children: bool = False) -> None:
```

Comment:
DESIGN: IMO this can be easily simplified.  It is **very important for maintainability** to remain close to the standard library specially when it comes to common functionality as removing a directory.

Suggested changes:

- Using shell commands **should be avoided** at all costs in python calls when there are python native functions that can do the job. IN this case, this function is [``shutil.rmtree``](https://docs.python.org/3/library/shutil.html#shutil.rmtree)
- To make it async you can simply [run it in an executor](https://docs.python.org/3/library/asyncio-eventloop.html#asyncio.loop.run_in_executor) . See the idea for instance in the implementation of ``aioshutil`` [here](https://en.wikipedia.org/wiki/Glob_(programming)). Or simply install ``aioshutil``

Reply:
check this https://github.com/pcrespov/sandbox-python/tree/python-stuff/lib-aiofiles

Reply:
Not a fan of it, but will remove the subprocess. Since I feel it is not to your liking.